### DISCUSSION TITLE
No transient model for microstrip lines

### DISCUSSION THREAD

**Discussion by goranmarinkovic** [ra3xdh/qucs_s#681] [Q&A] [answered]
I tried to simulate following design and do not get any output.
![image](https://github.com/ra3xdh/qucs_s/assets/86667287/f7f77739-1f61-450c-8187-0250f23e8d54)

The pulse generated is not visible on the out line and as well the cpld line is quiet... this does not look right.
The test files are attached for further investigation.
[breaking_example.zip](https://github.com/ra3xdh/qucs_s/files/14932712/breaking_example.zip)

**Comment by ra3xdh** [comment, marked as answer]
It is not a bug. Qucsator never had transient model for MS lines. The MS lines are considered as zero resistnace for all analysis except SP.

**Comment by ra3xdh** [comment]
The Qucsator from Qucs-0.0.20 gives the same result.

**Comment by goranmarinkovic** [comment]
Sorry I mixed QUCS-S and QucsStudio... there is see:
![image](https://github.com/ra3xdh/qucs_s/assets/86667287/9e0d951d-972a-4bcd-bb92-083d8bb7d37d)

**Comment by ra3xdh** [comment]
QucsStudio is closed source project not related to Qucs-S and may have incompatible extensions.

**Comment by goranmarinkovic** [comment]
It looks like a missing feature... since if it is possible in QucsStudio it should be available as well for Qucs-S.
It would be a pity if such an important feature is missing.

**Comment by ra3xdh** [comment]
Adding transient models for MS lines is technically possible if someone will provide equations.

**Comment by goranmarinkovic** [comment]
... here we need an RF expert. Anyone in the community able and willing to share?

**Comment by ra3xdh** [comment]
I have converted this issue to discussion. Probably someone could look at this and point a solution.

**Reply by goranmarinkovic** [reply]
I asked the RF expert in my company and he told me to point you to the Qucs-Studio documentation chapter 6.8 for the algorithm.

**Comment by tomhajjar** [comment]
@goranmarinkovic

You can use the ngspice TLine or Spice model of the needed device to make most transmission line elements like a coupler. See attached examples.

@ra3xdh

Do you have the BFP193W amplifier schematic used in this example?
![amp-ms](https://github.com/ra3xdh/qucs_s/assets/102838604/9cd8daf3-7107-4637-9418-3242b7bc28b5)

![2024-04-10_141440](https://github.com/ra3xdh/qucs_s/assets/102838604/844c0235-d64e-4a72-b40e-9c175b037647)
![2024-04-10_141418](https://github.com/ra3xdh/qucs_s/assets/102838604/87037c6f-44fb-4e89-88a9-69c7f5207c3b)

[BFP193W_amplifier_prj.zip](https://github.com/ra3xdh/qucs_s/files/14936027/BFP193W_amplifier_prj.zip)
[Transmission_Line_prj.zip](https://github.com/ra3xdh/qucs_s/files/14936028/Transmission_Line_prj.zip)

**Reply by goranmarinkovic** [reply]
Hello Tom,

Yes, I already used the TLINE for simulating microstrips in combination with the IBIS models using ngspice... this is now possible and would be a good extension for Qucs-S as well... since KiCAD does the translation of the IBIS model to spice models, such an extension looks feasible and would be useful for CPU/FPGAs, memory and MGT designs running at several GHz.
More info and examples you may find in this discussion here:  https://forum.kicad.info/t/ibis-models-with-fields-of-0-0-produce-simulation-errors/49982 
...simply download the working example and replace the 50 Ohm resistor by TLINE... and you're at the solution which I'm using currently.
Concerning your proposal: Sure, in frequency domain, that could be a workaround... what I'm interested in is PCB design... and there the possibility to add PCB dimensions before the HW is build has a practical benefit which I'm currently exploring for my projects... your proposal by S-parameters means just one step more, by creating the s-parameters for the coupled microstrip or stripline. Possible, but as I said one step more and error prone since I have to change the PCB dimensions during a project and you have to keep the S params in sync with the PCB dimensions. 

Question to you: Would the proposed implementation of MS in transition simulation be something you could use as well? I'm wondering, if I'm the only one preferring such a solution where I'm able to add the PCB dimensions and simulate the effects of design compromises.

Cheers
Goran

**Comment by tomhajjar** [comment]
I looked at that KiCAD example when I was troubleshooting issues with using TLine in Qucs-S. Unfortunately there seems to be few IBIS models available and the analysis only accounted for input/output "parameters" of digital gates.

As far as PCB design, trying to accurately replicate PCB traces and vias on a multi-layer PCB in "Qucs" doesn't seem feasible. You would be hard pressed to make an accurate schematic or find IBIS models for the desired devices. Don't you really need a 3D EM solver that would take in the actual PCB layout?

#465

**Comment by goranmarinkovic** [comment]
Hello Tom,

The IBIS models are in deed for digital electronics and MGTs as used for example for FPGAs (https://www.xilinx.com/support/download/index.html/content/xilinx/en/downloadNav/device-models/ibis-models/virtex-series-fpgas.html) or DDR SDRAMs (https://www.micron.com/search-results?searchRequest=IBIS) or many other digital components used up to few GHz. With many companies delivering exactly this IBIS model for the signal integrity checks... not more and not less.

The main purpose if the IBIS models is stated as follows:
"Traditional IBIS models are generally used instead of SPICE models to perform various board level signal integrity (SI) simulations and timing analyses. IBIS models could be used to verify signal integrity requirements, especially for high-speed products."

Hence, the industry does not need to publish the functionality of a component like in SPICE models but solely the IO buffer structure. This IBIS model combined with a model of the traces are used to simulate PCB designs and should potentially allow us to develop stable working PCBs.

An EM solver like ADS or openEMS are as well possible for making "precise" models of the traces, but the simulation effort is very high and computing power as well. Here the more simple ngspice and other simulators come into play.

Cheers
Goran

**Comment by tomhajjar** [comment]
Seems like a massive amount of work for a feature not many would use. What
actually is the goal?

Having extensively used QucsStudio, its transient analysis capability like
Qucs is abysmal. Also the inability to use most device Spice models limits
what devices are used. What circuits/devices do you want to analyze?

**Comment by goranmarinkovic** [comment]
@tomhajjar 

The goal is still to be able to simulate the coupling between adjacent tracks on a PCB. A regular PCB designer / electronics engineer can not use ADS or openEMS but he/she is able to see the effects of the coupling and do counter measures like separation, guard wires, serial termination at the source or the proper termination at the drain of the signal. QucsS could be the tool bridging the functional simulation between SPICE and EM solvers.
I guess there are more PCB designer / electronics engineers world wide in need for such an analysis than RF engineers doing the precise design of a "simple" digital PCB. However, the current approach to build a PCB and then measure is not up to the art of engineering. 

Cheers
Goran

**Comment by tomhajjar** [comment]
You're assuming the models used in Qucs are accurate AND more importantly
you could make a schematic that accurately models PCB traces/coupled lines
and multiple types of vias meandering on a multi-layer PCB.

**Comment by goranmarinkovic** [comment]
@tomhajjar 
To be more on the true side: "I assume the models used in Qucs are accurate **enough**..." Concerning the coupled striplines or microstrips. I know the PCB geometry and manufacturing tolerances good enough to do the input to QucsS. Please note, that for a common digital electronics engineer the perfect signal is most of the time not relevant as long as he is able to distinguish reliably a 0 from a 1 :-)

**Comment by ra3xdh** [comment]
I have created https://github.com/ra3xdh/qucsator_rf/issues/7 The adding MS lines transient models for QucsatorRF is planned for the near future. Help is highly appreciated. I can provide a necessary guidance.

**Comment by ra3xdh** [comment]
I have started with implementation of the TRAN model for microstrip lines etc. The help from RF experts is appreciated. See https://github.com/ra3xdh/qucsator_rf/pull/16
